templates/assets/plugins/image/popcorn.image.js

```
@@ -23,7 +24,30 @@
   }
 
   function getPhotoSet( photosetId , ready ) {
-    var uri = getPhotosetCmd + "&photoset_id=" + photosetId + APIKEY + jsonBits;
+    var photoSplit,
+        uri;
+
+    /* Allow for a direct gallery URL to be passed or just a gallery ID. This will accept:
+     *
+     * http://www.flickr.com/photos/etherworks/sets/72157630563520740/
+     * or
+     * 72157630563520740
+     */
+    if ( isNaN( photosetId ) ) {
+      photosetId = photosetId.substring( photosetId.indexOf( "www." ) );
+      photoSplit = photosetId.split( "/" );
+
+      // Can't always look for the ID in the same spot depending if the user includes the
+      // last slash
+      for ( var i = 0; i < photoSplit.length; i++ ) {
```

What happens if I try to use a non-flickr url here? The editor should probably report an error saying "use valid flickr urls or gallery ID's".